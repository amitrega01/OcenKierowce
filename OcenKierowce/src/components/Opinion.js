import React, { Component } from 'react';
import { TouchableOpacity, Text, StyleSheet } from 'react-native';

class Opinion extends Component {
  constructor(props) {
    super(props);
  }
  render() {
    var color;
    switch (this.props.opinion.type) {
      case 'UPVOTE': {
        color = '#02e053';
        break;
      }

      case 'DOWNVOTE': {
        color = '#ed0000';
        break;
      }
    }
    return (
      <TouchableOpacity
        activeOpacity={0.8}
        style={[styles.wrapper, { backgroundColor: color }]}
        onPress={this.props.onPress}>
        <Text style={styles.text}>{this.props.opinion.message}</Text>
      </TouchableOpacity>
    );
  }
}

export default Opinion;
const styles = StyleSheet.create({
  wrapper: {
    margin: 8,
    justifyContent: 'center',
    alignItems: 'center',
    padding: 16,
    elevation: 4,
    borderRadius: 20,
    padding: 20,
  },
  text: {
    fontWeight: 'normal',
    color: '#fff',
    fontSize: 15,
  },
});
