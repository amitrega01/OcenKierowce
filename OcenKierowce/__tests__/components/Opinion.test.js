import React from 'react';
import renderer from 'react-test-renderer';

import Opinion from '../../src/components/Opinion';

describe('<Opinion />', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Opinion
          opinion={{
            author: {
              anonymouns: true,
              uid: 'qVZHotniHyXWRbLBoXBeVaNs3jf2',
            },
            message: 'Wiadomosc',
            photo: true,
            plateNumber: 'SCI79308',
            region: 'śląskie',
            timeStamp: 1559207790870,
            type: 'DOWNVOTE',
            _id: '-Lg70giVOnhIWZIe9WkM',
          }}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
