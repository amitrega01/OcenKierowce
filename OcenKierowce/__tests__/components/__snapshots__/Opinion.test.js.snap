// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Opinion /> renders correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#ed0000",
      "borderRadius": 20,
      "elevation": 4,
      "justifyContent": "center",
      "margin": 8,
      "opacity": 1,
      "padding": 20,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#fff",
        "fontSize": 15,
        "fontWeight": "normal",
      }
    }
  >
    Wiadomosc
  </Text>
</View>
`;
